src/js/dashboard/dashboard.controller.js

```diff
@@ -23,6 +23,11 @@ function DashboardComponent(franklinAPIService, $scope,
 
   dc.getFranklinRepos();
 
+  //watch changes in franklin repos model
+  $scope.$watch(franklinReposModel.getFranklinRepos,
```

Comment:
franklinReposModel is the list of Franklin repos in the system and this list is modified by more than one controller, so we need to watch this as long as the app is running for consistency between the different pages so we can add/delete only the repo that changed instead of making another call to the server to retrieve the whole list each time.